services/debugger/cache/cache.go

```diff
@@ -0,0 +1,46 @@
+package cache
+
+//go:generate mockgen -destination=../../../mocks/services/debugger/cache/mock_cache.go -package=mocks_cache github.com/rudderlabs/rudder-server/services/debugger/cache CacheAny
+
+import (
+	"github.com/rudderlabs/rudder-server/services/debugger/cache/embeddedcache"
+	"github.com/rudderlabs/rudder-server/services/debugger/cache/listcache"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+)
+
+var pkgLogger logger.Logger
+
+type CacheAny interface {
+	Cache[any]
+}
```

Comment:
What is the purpose of this interface? It doesn't seem to be used.

Reply:
This is unused but the gomock we use does not seem to be having the capability to generate mocks for generic interfaces and hence this hack

Reply:
Then I'd say don't use gomock :slightly_smiling_face:
It seems very simple to mock without assistance from a library :man_shrugging:

Reply:
Got rid of the interface and the mock